object-sync-for-salesforce.php

```diff
@@ -167,40 +170,71 @@ protected function __construct() {
 
-		$this->logging = $this->logging( $this->wpdb, $this->version );
+		$this->activated = $this->activate( $this->wpdb, $this->version, $this->slug, $this->option_prefix, $this->schedulable_classes );
+		$this->deactivate( $this->wpdb, $this->version, $this->slug, $this->option_prefix, $this->schedulable_classes );
 
-		$this->mappings = $this->mappings( $this->wpdb, $this->version, $this->slug, $this->logging );
+		$this->logging = $this->logging( $this->wpdb, $this->version, $this->slug, $this->option_prefix );
 
-		$this->wordpress  = $this->wordpress( $this->wpdb, $this->version, $this->slug, $this->mappings, $this->logging );
+		$this->mappings = $this->mappings( $this->wpdb, $this->version, $this->slug, $this->option_prefix, $this->logging );
+
+		$this->wordpress  = $this->wordpress( $this->wpdb, $this->version, $this->slug, $this->option_prefix, $this->mappings, $this->logging );
 		$this->salesforce = $this->salesforce_get_api();
 
-		$this->push = $this->push( $this->wpdb, $this->version, $this->login_credentials, $this->slug, $this->wordpress, $this->salesforce, $this->mappings, $this->logging, $this->schedulable_classes );
+		$this->push = $this->push( $this->wpdb, $this->version, $this->login_credentials, $this->slug, $this->option_prefix, $this->wordpress, $this->salesforce, $this->mappings, $this->logging, $this->schedulable_classes, $this->queue );
 
-		$this->pull = $this->pull( $this->wpdb, $this->version, $this->login_credentials, $this->slug, $this->wordpress, $this->salesforce, $this->mappings, $this->logging, $this->schedulable_classes );
+		$this->pull = $this->pull( $this->wpdb, $this->version, $this->login_credentials, $this->slug, $this->option_prefix, $this->wordpress, $this->salesforce, $this->mappings, $this->logging, $this->schedulable_classes, $this->queue );
 
-		$this->load_admin( $this->wpdb, $this->version, $this->login_credentials, $this->slug, $this->wordpress, $this->salesforce, $this->mappings, $this->push, $this->pull, $this->logging, $this->schedulable_classes );
+		$this->load_admin( $this->wpdb, $this->version, $this->login_credentials, $this->slug, $this->option_prefix, $this->wordpress, $this->salesforce, $this->mappings, $this->push, $this->pull, $this->logging, $this->schedulable_classes, $this->queue );
 
 	}
 
+	/**
+	 * Load immediately required things
+	 *
+	 * @param object $wpdb
+	 * @param string $version
+	 * @param string $slug
+	 * @param string $option_prefix
+	 *
+	 */
+	private function load( $wpdb, $version, $slug, $option_prefix ) {
+		require_once plugin_dir_path( __FILE__ ) . 'vendor/autoload.php';
+	}
+
+	/**
+	 * Get queue instance.
+	 *
+	 * @param object $wpdb
+	 * @param string $version
+	 * @param string $slug
+	 * @param string $option_prefix
+	 * @param array $schedulable_classes
+	 * @return Object_Sync_Sf_Queue
+	 */
+	public function queue( $wpdb, $version, $slug, $option_prefix, $schedulable_classes ) {
```

Comment:
The rest of the methods (logging, mappings, wordpress, etc) are all private. Is there a reason this one is public?